### PR TITLE
Setup framework interface

### PR DESCRIPTION
**Pull request by qiujian16**

### CODE REVIEW

File: pkg/agent/inteface.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+	// GetAgentAddonOptions returns the agent options.
+	GetAgentAddonOptions() AgentAddonOptions
+}
+
+// AgentAddonWithRegistration manifests of agent deployed on managed cluster, rbac rules and bootstrap secret for agent to register to hub.
+type AgentAddonWithRegistration interface {
+	AgentAddon
+	// HubRBAC returns the role and rolebinding on the hub for an agent on the managed cluster.
+	HubRBAC(cluster *clusterv1.ManagedCluster, group string) (*rbacv1.Role, *rbacv1.RoleBinding)
```
Comment (elgnay) [inline pkg/agent/inteface.go]: Do we need to support clusterrole/clusterrolebindnig?

Comment (qiujian16) [reply pkg/agent/inteface.go]: added

File: cmd/helloworld/main.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+func (h *helloWorldAgent) GetRegistrationOption() *agent.RegistrationOption {
+	// return nil since registration is not needed by this agent
+	return nil
+}
+
+// Another agent with registration enabled.
+type helloWorldAgentWithRegistration struct{}
+
+func (h *helloWorldAgentWithRegistration) Manifests(cluster *clusterv1.ManagedCluster) ([]runtime.Object, error) {
+	return []runtime.Object{
```
Comment (deads2k) [inline cmd/helloworld/main.go]: I think you may want to show how to use yaml files read in as well.  That gives people the option of describing it both ways.

Comment (qiujian16) [reply cmd/helloworld/main.go]: yes, I added an example to use  yaml  files.

**Pull request merged**